SimplifiedTests/NYPLAxisTests/NYPLAxisItemDownloaderTests.swift

```diff
@@ -198,4 +203,71 @@ class NYPLAxisItemDownloaderTests: XCTestCase {
     waitForExpectations(timeout: 4, handler: nil)
   }
   
+  func testDownloaderShouldNotUpdateProgressUponFailureDownloadingAsset() {
+    let dispatchGroup = DispatchGroup()
+    let assetWriter = AssetWriterMock().mockingSuccess()
+    
+    let itemDownloader = NYPLAxisItemDownloader(
+      assetWriter: assetWriter, dispatchGroup: dispatchGroup,
+      downloader: contentDownloader, weightProvider: weightProvider)
+    
+    let progressListener = NYPLAxisProgressListenerMock()
+    itemDownloader.delegate = progressListener
+    XCTAssertEqual(progressListener.currentProgress, 0)
+    
+    dispatchGroup.enter()
```

Comment:
would you say this usage of the dispatchGroup assumes knowledge of how the group is used inside the  code being tested?  Or is this documented? it probably is/should, but I don't recall. From this point of view I wouldn't know why I need to "enter" the group here and wait later, for example. If it's already documented in NYPLAxisItemDownloader, this is fine.

Reply:
You're right. We're entering dispatchGroup from multiple places. However, NYPLAxisItemDownloader should be the only place where we enter and leave the dispatchGroup and other classes should not be aware of it. Added a TODO note in code. Will work on that.